packages/sdk-codegen/src/codeGen.ts

```diff
@@ -235,7 +239,7 @@ export abstract class CodeGen implements ICodeGen {
         )
         propertyValues = props.join(this.enumDelimiter)
       } else {
-        Object.values(type.properties).forEach((prop) =>
+        this.typeProperties(type).forEach((prop) =>
```

Comment:
this was my solution to allow custom behavior for language generators and keep the previous behavior the default. open to other ways if this doesn't seem right.

Reply:
I think this looks reasonable